src/main/sensors/compass.c

```diff
@@ -375,23 +391,34 @@ bool compassIsCalibrationComplete(void)
 
 uint32_t compassUpdate(timeUs_t currentTimeUs)
 {
-    static uint8_t busyCount = 0;
-    uint32_t nextPeriod = COMPASS_READ_US;
-
-    if (busBusy(&magDev.dev, NULL) || !magDev.read(&magDev, magADCRaw)) {
-        // No action was taken as the read has not completed
-        schedulerIgnoreTaskStateTime();
-
-        busyCount++;
-
-        return nextPeriod; // Wait COMPASS_READ_US between states
+    bool checkBusBusy = busBusy(&magDev.dev, NULL);
+    bool checkReadState = !magDev.read(&magDev, magADCRaw);
```

Comment:
Old code did skip `read` on busy, probably intentionally ...

Reply:
Good point, thank you.
Perhaps commit [54866d](https://github.com/betaflight/betaflight/pull/13166/commits/548666df4f9ee0d9a034d64f196b986c0c6ab6a1) might give the original behaviour?